worker/jwtparser/manifold.go

```diff
@@ -0,0 +1,66 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package jwtparser
+
+import (
+	"github.com/juju/errors"
+	"github.com/juju/worker/v3"
+	"github.com/juju/worker/v3/dependency"
+
+	workerstate "github.com/juju/juju/worker/state"
+)
+
+// ManifoldConfig defines a Manifold's dependencies.
+type ManifoldConfig struct {
+	StateName string
+}
+
+// Manifold returns a manifold whose worker wraps a JWT parser.
+func Manifold(config ManifoldConfig) dependency.Manifold {
+	inputs := []string{config.StateName}
+	return dependency.Manifold{
+		Inputs: inputs,
+		Output: outputFunc,
+		Start:  config.start,
+	}
+}
+
+// start is a method on ManifoldConfig because it's more readable than a closure.
+func (config ManifoldConfig) start(context dependency.Context) (worker.Worker, error) {
+	var stTracker workerstate.StateTracker
+	if err := context.Get(config.StateName, &stTracker); err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	statePool, err := stTracker.Use()
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	defer func() {
+		_ = stTracker.Done()
+	}()
```

Comment:
You can't defer here, as this will cause use after free errors.

You need to use a cleanup worker function for this.